### DISCUSSION TITLE
Add support for environment variables/secrets in Flows

### DISCUSSION THREAD

**Discussion by colemahatmccreary** [directus/directus#13948] [Feature Requests]
Pretty straightforward concept to safely store/access credentials and other sensitive data.

For instance, I want to dispatch a webhook from the flow every time a customer's profile is updated, and would rather not hard code the relevant access tokens every time.

Probably a few things to think about here as the sensitive data might still become visible in the flow's logs, but some level of obfuscation would be nice (at least within the 'build flow' GUI).

Potentially related topic: https://github.com/directus/directus/discussions/10591

**Comment by NeoKactus** [comment]
I was thinking how this could be implemented with a custom hook. ~~(I am assuming we can access ` env from '../env';` in the custom hook)~~

Edit: Had trouble finding it, but for env access you can actually get it when you define your hook! 
 `export default defineHook(({ filter }, { services, exceptions, env }) => {}`

**Steps:**
- **Custom hook**
  - Add a json/longtext `env` field to `directus_settings`.
  - Create a custom action hook that triggers from the `server.start` event.
  - Update the `env` field with a whitelist of `env` variables you want.

- **Get ENV Flow** (Unsure about this part. I am assuming `Another Flow` triggers can be used to created reusable logic)
  - Create a `Get ENV` flow with `Another Flow` as the trigger.
  - Add a `Read Data` operation that grabs `env` from `directus_settings`

- **Original Flow**
  - Use the `Trigger Flow` operation to call `Get ENV` flow. (Unsure)
  - Or just do the `Read Data` operation to grab the `env` variables.

Haven't tested any of this, but was curious with playing around with the idea. That being said, a more native solution that avoids having to use a `Get ENV` operation would be much more ergonomic/secure.

**Reply by benhaynes** [reply]
Thank you for adding this, @King-Aardvark... this discussion is important and seems to be getting some fast traction in terms of votes!

**Comment by nickrum** [comment]
Env access in flows through `$env` has been added as part of #16111.

**Reply by copostic** [reply]
Thanks for this, however it's still only available for Run script, not for other Operations :(
Is it scheduled ?

**Reply by nickrum** [reply]
Just checked and env vars are definitely available under the `$env` key by using `{{ $env.<env-var-name> }}` in any of the operations. Make sure you also set `FLOWS_ENV_ALLOW_LIST` to the list of env var names you want to expose to flows.
